File: jan16/jan16.js
function decimalToBinary(decimalNumber) {
    return decimalNumber.toString(2);
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question('Enter a decimal number: ', (input) => {
        const decimalNumber = parseInt(input);
        if (isNaN(decimalNumber) || decimalNumber < 0) {
            console.log("Please enter a valid non-negative integer.");
        } else {
            const binaryRepresentation = decimalToBinary(decimalNumber);
            console.log(`The binary representation of ${decimalNumber} is: ${binaryRepresentation}`);
        }
        rl.close();
    });
}

main();

File: jan16/jan16.java
import java.util.Scanner;

public class DecimalToBinary {
    public static String decimalToBinary(int decimalNumber) {
        return Integer.toBinaryString(decimalNumber);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a decimal number: ");
        int decimalNumber = scanner.nextInt();

        if (decimalNumber < 0) {
            System.out.println("Please enter a non-negative integer.");
        } else {
            String binaryRepresentation = decimalToBinary(decimalNumber);
            System.out.println("The binary representation of " + decimalNumber + " is: " + binaryRepresentation);
        }
        scanner.close();
    }
}